statsmodels/regression/heckman.py

```diff
@@ -0,0 +1,666 @@
+"""
+Heckman correction for sample selection bias (the Heckit procedure).
+
+Created August 19, 2014 by B.I.
+Last modified September 02, 2014 by B.I.
+
+NO warranty is provided for this software.
+"""
+
+import numpy as np
+import statsmodels.api as sm
+import statsmodels.base.model as base
+from statsmodels.iolib import summary
+from statsmodels.tools.numdiff import approx_fprime
+from scipy.stats import norm
+from scipy.stats import t
+import warnings
+import pandas as pd
+import pdb
+
+class Heckman(base.LikelihoodModel):
+    """
+    Class for Heckman correction for sample selection bias model.
+
+    Attributes
+    ----------
+    endog : 1darray
+        Data for the dependent variable. Should be set to np.nan for
+        censored observations.
+    exog : 2darray
+        Data for the regression (response) equation. If a constant
+        term is desired, the user should directly add the constant
+        column to the data before using it as an argument here.
+    exog_select : 2darray
+        Data for the selection equation. If a constant
+        term is desired, the user should directly add the constant
+        column to the data before using it as an argument here.
+    **kwargs:
+        missing=, which can be 'none', 'drop', or 'raise'
+
+    Notes
+    -----
+    The selection equation should contain at least one variable that 
+    is not in the regression (response) equation, i.e. the selection
+    equation should contain at least one instrument. However, this
+    module should still work if the user chooses not to do this.
+    """
+
+    def __init__(self, endog, exog, exog_select, **kwargs):
```

Comment:
Ok found it. I was thinking too much about treatment effects models that I forgot that it's the selection effect. Stata has it optional whether depvar for selection is specified or not.